pkg/imgpkg/registry/keychain_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package registry_test
+
+import (
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/k14s/imgpkg/pkg/imgpkg/registry"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAnonAuthWhenNoEnvVarsProvided(t *testing.T) {
+	keychain := registry.Keychain(registry.KeychainOpts{}, func() []string { return nil })
+	resource, err := name.NewRepository("some.fake.registry/imgpkg_test")
+	assert.NoError(t, err)
+
+	auth, err := keychain.Resolve(resource)
+	assert.NoError(t, err)
+
+	assert.Equal(t, authn.Anonymous, auth)
+}
+
+func TestEnvAuthWhenEnvVarsProvided(t *testing.T) {
```

Comment:
Can we add in one of these tests an environment variable that looks like "SOMETHING_USERNAME" to ensure we are only considering the ones that start with `IMGPKG_REGISTRY`?

Reply:
Ok done